extensions/Worker.Extensions.Storage.Blobs/src/BlobStorageConverter.cs

```diff
@@ -90,12 +93,15 @@ private async ValueTask<ConversionResult> ConvertFromCollectionBindingDataAsync(
 
                     if (element is not null)
                     {
-                        blobCollection.Add(element);
+                        result[i] = element;
                     }
                 }
 
-                var methodName = context.TargetType.IsArray ? nameof(CloneToArray) : nameof(CloneToList);
-                var result = ToTargetTypeCollection(blobCollection, methodName, elementType);
+                if (!context.TargetType.IsArray)
+                {
+                    var resultType = typeof(List<>).MakeGenericType(elementType);
+                    result = (IList)Activator.CreateInstance(resultType, result);
```

Comment:
This is pretty cool, learnt something new :)

Reply:
+1. Nice approach!